Parse/src/main/java/com/parse/Parse.java

```diff
@@ -35,6 +36,86 @@
  * library.
  */
 public class Parse {
+  public static final class Configuration {
+    private final Context context;
+    private final String applicationId;
+    private final String clientKey;
+    private final boolean localDataStoreEnabled;
+    private final List<ParseNetworkInterceptor> interceptors;
+
+    private Configuration(Builder builder) {
+      this.context = builder.context;
+      this.applicationId = builder.applicationId;
+      this.clientKey = builder.clientKey;
+      this.localDataStoreEnabled = builder.localDataStoreEnabled;
+      this.interceptors = builder.interceptors;
+    }
+
+    public static final class Builder {
+      private Context context;
+      private String applicationId;
+      private String clientKey;
+      private boolean localDataStoreEnabled;
+      private List<ParseNetworkInterceptor> interceptors = new ArrayList<>();
+
+      public Builder(Context context) {
+        this.context = context;
+      }
```

Comment:
This will be a pain for us if we ever want to add vanilla Java support, but we can't really get around it and the problem exists within it's parent class already.